src/components/ItemList.svelte

```diff
@@ -18,22 +20,41 @@
   export let totalItems;
   export let paginatedItems;
 
+  let nonExpiredItems = items.filter((item) => !isExpired(item.expires));
+  let shownItems = nonExpiredItems;
+
+  let showExpired = false;
+  let filterTerm = "";
+
   let currentPage = writable(1);
   setContext("currentPage", currentPage);
 
-  totalItems = items.length;
-  paginatedItems = chunk([...items], DEFAULT_ITEMS_PER_PAGE);
+  totalItems = shownItems.length;
+  paginatedItems = chunk([...shownItems], DEFAULT_ITEMS_PER_PAGE);
 
   function filterItems(filterText) {
-    filteredItems = items.filter((item) => item.name.includes(filterText));
+    filterTerm = filterText;
+    filteredItems = shownItems.filter((item) => item.name.includes(filterText));
     if (filteredItems.length > 0) {
       currentPage.set(1);
       paginatedItems = chunk([...filteredItems], DEFAULT_ITEMS_PER_PAGE);
       totalItems = filteredItems.length;
     }
   }
 
-  $: filteredItems = paginatedItems[$currentPage - 1];
+  function handleExpiredItems(checked) {
```

Comment:
Should use a more descriptive variable name here:

```suggestion
  function handleExpiredItems(showExpired) {
```

I feel like we should somehow combine this function with filterItems above, to generate the list of shownItems. Is that possible? It would be nice if we could simplify this component a bit.